src/main/java/edu/harvard/iq/dataverse/api/Files.java

```diff
@@ -818,4 +829,50 @@ public Response getExternalToolFMParams(@Context ContainerRequestContext crc, @P
     public Response getFixityAlgorithm() {
         return ok(systemConfig.getFileFixityChecksumAlgorithm().toString());
     }
+
+    @GET
+    @AuthRequired
+    @Path("{id}/guestbookResponses/count")
+    public Response getCountGuestbookResponses(@Context ContainerRequestContext crc, @PathParam("id") String dataFileId) {
+        return response(req -> {
+            DataFile dataFile = execCommand(new GetDataFileCommand(req, findDataFileOrDie(dataFileId)));
+            return ok(guestbookResponseService.getCountGuestbookResponsesByDataFileId(dataFile.getId()).toString());
+        }, getRequestUser(crc));
+    }
+
+    @GET
+    @AuthRequired
+    @Path("{id}/canBeDownloaded")
+    public Response canFileBeDownloaded(@Context ContainerRequestContext crc, @PathParam("id") String dataFileId) {
+        DataFile dataFile;
+        try {
+            dataFile = findDataFileOrDie(dataFileId);
+        } catch (WrappedResponse wr) {
+            return wr.getResponse();
+        }
+        return ok(fileDownloadServiceBean.canDownloadFile(getRequestUser(crc), dataFile.getFileMetadata()));
+    }
```

Comment:
Permission checks are carried out inside the canDownloadFile method. There would be no problem in the scenario you mention, as long as the the file is downloadable by the Guest User. In fact, that a guest user can download files is a possible scenario.

FWIW - I suggest reading my note on this particular endpoint in the PR description

Reply:
I'm just noting that we at least talked about the idea of requesting true/false for 10 files at once. This might be a performance problem.

Reply:
I agree that the fact that the new API is called on one file at a time has a potential for becoming a performance issue on a slow connection. That was my primary motivation behind suggesting to (optionally) perform these checks as part of the  `datasets/{id}/versions/{versionId}/files` api and adding a "canDownload" flag to its output... But the fact that it will always be for only 1 page-worth of files at a time probably makes it ok to leave that for another PR. We should be able to find out experimentally whether this will be significant or noticeable. It should indeed be easy to add a version of this API that takes a list of filemetadata ids.